core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala

```diff
@@ -0,0 +1,138 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.h2o.backends.external
+
+import org.apache.spark.h2o.H2OConf
+import org.apache.spark.h2o.backends.SharedH2OConf
+
+/**
+  * External backend configuration
+  */
+trait ExternalBackendConf extends SharedH2OConf {
+  self: H2OConf =>
+
+  import ExternalBackendConf._
+
+  def h2oCluster = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+  def YARNQueue = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_YARN_QUEUE._1)
+  def h2oDriverPath = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_H2O_DRIVER._1)
+  def numOfExternalH2ONodes = sparkConf.getOption(PROP_EXTERNAL_H2O_NODES._1)
+  def HDFSOutputDir = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_HDFS_DIR._1)
+  def mapperXmx = sparkConf.get(PROP_EXTERNAL_H2O_MEMORY._1, PROP_EXTERNAL_H2O_MEMORY._2)
+
+
+  /**
+    * Sets node and port representing H2O Cluster to which should H2O connect when started in external mode.
+    * This method automatically sets external cluster mode
+    *
+    * @param host host representing the cluster
+    * @param port port representing the cluster
+    * @return H2O Configuration
+    */
+  def setH2OCluster(host: String, port: Int): H2OConf = {
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, host + ":" + port)
+    self
+  }
+
+  def setH2OCluster(hostPort: String): H2OConf = {
+    setExternalClusterMode()
+    sparkConf.set(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1, hostPort)
+    self
+  }
+
+  def h2oClusterHost = {
+    val value = sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1)
+    if (value.isDefined) {
+      Option(value.get.split(":")(0))
+    } else {
+      None
+    }
```

Comment:
This is not Scala style. The right Scala way is a one-liner:

sparkConf.getOption(PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1) map (_.split(":")(0))